### PR TITLE
refactor(fe): refactor handle query

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
some refactor on handle query

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by st1page** [comment]
this PR will change the meaning of the frontend's metrics latency_local_execution. 
- past latency_local_execution = planning time + execute time
- now  latency_local_execution = execute time

File: src/frontend/src/handler/query.rs [hunk 1]
```diff
@@ -12,61 +12,103 @@
...
-            if stmt_type.is_dml() {
-                // DML do not support local mode yet.
-                distribute_execute(context, bound, format).await?
-            } else {
-                local_execute(context, bound, format).await?
-            }
+    let mut planner = Planner::new(context);
+
+    let must_local = force_local_mode(&bound);
+    let must_dist = stmt_type.is_dml();
```
Comment (liurenjie1024) [inline src/frontend/src/handler/query.rs:56]: Why not put this logic in `force_local_mode`?

File: src/frontend/src/handler/query.rs [hunk 2]
```diff
@@ -12,61 +12,103 @@
...
+    format: bool,
+) -> Result<RwPgResponse> {
+    let stmt_type = to_statement_type(&stmt);
+    let session = context.session_ctx.clone();
+
+    // Subblock to make sure PlanRef (an Rc) is dropped before `await` below.
+    let (query, query_mode, pg_descs) = {
+        let (plan, query_mode, pg_descs) = gen_batch_query_plan(&session, context.into(), stmt)?;
+
+        tracing::trace!(
```
Comment (liurenjie1024) [inline src/frontend/src/handler/query.rs:95]: Remove `distributed`?

File: src/frontend/src/handler/query.rs (hunk 1 above)
Comment (st1page) [reply src/frontend/src/handler/query.rs:56]: I will remove the `force_local_mode` because now this is the only place using it :smile_cat:

**Comment by liurenjie1024** [comment]
I don't think we need to change the meaning of local execution latency, and a little refactoring to `gen_batch_plan` method can fix it.

**Review by BowenXiao1999 — APPROVED** [review]

**Comment by st1page** [comment]
>  I don't think we need to change the meaning of local execution latency, and a little refactoring to gen_batch_plan method can fix it.

fixed

File: src/frontend/src/handler/query.rs [hunk 3]
```diff
@@ -154,15 +150,23 @@ pub async fn handle_query(
 
     // update some metrics
     if query_mode == QueryMode::Local {
-        latency_local_execution_timer.observe_duration();
+        fn duration_to_seconds(d: Duration) -> f64 {
```
Comment (liurenjie1024) [inline src/frontend/src/handler/query.rs:153]: Why not use
https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.as_secs_f64 ?

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Pull request merged**